cmd/gitops/check/cmd.go

```diff
@@ -0,0 +1,64 @@
+package check
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/weaveworks/weave-gitops/pkg/services/check"
+	"github.com/weaveworks/weave-gitops/pkg/version"
+
+	"github.com/weaveworks/weave-gitops/pkg/osys"
+	"github.com/weaveworks/weave-gitops/pkg/runner"
+
+	"github.com/weaveworks/weave-gitops/pkg/flux"
+
+	"github.com/spf13/cobra"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+)
+
+var (
+	pre bool
+)
+
+var Cmd = &cobra.Command{
+	Use:   "check",
+	Short: "Validates flux compatibility",
+	Example: `
+# Validate flux and kubernetes compatibility
+gitops check --pre
+`,
+	RunE: runCmd,
+}
+
+func init() {
+	Cmd.Flags().BoolVarP(&pre, "pre", "p", true, "perform only the pre-installation checks")
+}
+
+func runCmd(_ *cobra.Command, _ []string) error {
+	ctx := context.Background()
+
+	fluxClient := flux.New(osys.New(), &runner.CLIRunner{})
+
+	rest, clusterName, err := kube.RestConfig()
+	if err != nil {
+		return fmt.Errorf("failed getting rest config: %w", err)
+	}
+
+	_, k8sClient, err := kube.NewKubeHTTPClientWithConfig(rest, clusterName)
+	if err != nil {
+		return fmt.Errorf("failed creating k8s client: %w", err)
+	}
+
+	kubeClient := &kube.KubeHTTP{
+		Client: k8sClient,
+	}
+
+	output, err := check.Pre(ctx, kubeClient, fluxClient, version.FluxVersion)
```

Comment:
I am confused, I thought with marks comments that check should run more than just the pre checks. I was under the impression that if I run `gitops check` it checks everything and then if I add --pre it will only run the pre check. I am not seeing anywhere that the pre flag is even being used.

Reply:
Good question. This ticket is just about running pre-checks. Any other checks are out of the scope of this ticket.